packages/untp-test-suite/src/templates/utils.ts

```diff
@@ -0,0 +1,49 @@
+import { ErrorObject } from 'ajv';
+import {
+  IValidatedCredentials,
+  TemplateName,
+  FinalStatus,
+  ICredentialTestResult,
+  IFinalReport,
+  TestSuiteMessage,
+} from '../core/types/index.js';
+
+export const getTemplateName = (testSuiteResult: IValidatedCredentials): string => {
+  if (testSuiteResult.errors === null) {
+    return TemplateName.pass;
+  }
+
+  if (Array.isArray(testSuiteResult.errors) && testSuiteResult.errors.length > 0) {
+    const errorObjects = testSuiteResult.errors as ErrorObject[];
+    const errorWithAdditionalProperty = errorObjects.find((error) => error?.params?.additionalProperty);
+    if (errorWithAdditionalProperty) {
+      // If any ErrorObject has 'additionalProperty', return 'warning'
+      return TemplateName.warn;
+    }
+
+    // If no ErrorObject has 'additionalProperty', return 'FAIL'
+    return TemplateName.error;
+  } else {
+    return TemplateName.pass;
+  }
+};
```

Comment:
I think the first if statement should handle all cases to return the 'pass' template, and then the remaining code should only focus on returning the 'warn' or 'error' template.

This way, your code will be cleaner and easier to read. 

You can consider to this approach:

```ts
export const getTemplateName = (testSuiteResult: IValidatedCredentials): string => {
  if (!testSuiteResult.errors || !testSuiteResult.errors.length) {
    return TemplateName.pass;
  }

  const errors = testSuiteResult.errors as ErrorObject[];
  const isWarningType = errors.find((error) => error?.params?.additionalProperty);
  if (isWarningType) {
    // If any ErrorObject has 'additionalProperty', return 'warning'
    return TemplateName.warn;
  }

  // If no ErrorObject has 'additionalProperty', return 'FAIL'
  return TemplateName.error;
};
```